component/file_cache/file_cache.go

```diff
@@ -266,9 +270,27 @@ func (c *FileCache) OnConfigChange() {
 	c.cacheTimeout = float64(conf.Timeout)
 	c.policyTrace = conf.EnablePolicyTrace
 	c.offloadIO = conf.OffloadIO
+	c.maxCacheSize = conf.MaxSizeMB
 	c.policy.UpdateConfig(c.GetPolicyConfig(conf))
 }
 
+func (c *FileCache) StatFs() (*syscall.Statfs_t, bool, error) {
+	// cache_size = f_blocks * f_frsize/1024
+	// cache_size - used = f_frsize * f_bavail/1024
+	// cache_size - used = vfs.f_bfree * vfs.f_frsize / 1024
+	usage := getUsage(c.tmpPath) * MB
```

Comment:
If user has configured 0 (unlimited) cache limit then how are we calculating it ?

Reply:
in that case just use root mount stats since availability will depend on it.